django_refugeerights/subscription/views.py

```diff
@@ -13,3 +18,29 @@ class SubscriptionViewSet(viewsets.ModelViewSet):
     queryset = Subscription.objects.all()
     serializer_class = SubscriptionSerializer
     filter_fields = ('contact_key', 'to_addr', 'active', )
+
+
+class SwitchSubscriptionView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, format=None):
+        data = request.data
+        try:
+            contact_key = data['contact_key']
+        except KeyError:
+            raise ValidationError("A contact 'key' parameter is required.")
+        try:
+            messageset_id = data['messageset_id']
+        except KeyError:
+            raise ValidationError("A messageset 'id' parameter is required.")
+
+        messageset_exists = MessageSet.objects.filter(
+            pk=messageset_id).exists()
+        if not messageset_exists:
+            raise ValidationError("A messageset 'id' doesn't exist")
+
+        Subscription.objects.filter(
+            pk=contact_key, active=True, completed=False).update(active=False)
+        Subscription.save(messageset_id=messageset_id,
+                          messageset_exists=messageset_exists)
```

Comment:
I think you want to use `Subscription.objects.create(...)` here? Maybe read the Django documentation on models?